packages/react-components/react-tag-picker/library/src/utils/useExpandLabel.ts

```diff
@@ -0,0 +1,108 @@
+  const { tagPickerId, state } = options;
+  const { targetDocument } = useFluent();
+  const triggerRef = useTagPickerContext_unstable(ctx => ctx.triggerRef);
+  const expandIconRef = React.useRef<HTMLSpanElement>(null);
+
+  const expandIcon = { ...state.expandIcon };
+  const hasExpandIcon = !!state.expandIcon;
+  const {
+    'aria-label': expandIconAriaLabel,
+    'aria-labelledby': expandIconAriaLabelledby,
+    id: expandIconId,
+  } = expandIcon;
+
+  // If aria-label or aria-labelledby changes, recalculate aria-label and aria-labelledby for the expandIcon
+  // The expandIcon's label is calculated based on the input's label
+  // TODO: investigate ways to enforce client to provide a label rather than need to calculate it
+  const getExpandLabel = React.useCallback(
+    (ariaLabel?: string | null, ariaLabelledBy?: string | null) => {
+      let expandAriaLabel = undefined;
+      let expandAriaLabelledBy = undefined;
+      let expandId = undefined;
+
+      if (hasExpandIcon) {
+        const hasExpandLabel = expandIconAriaLabel || expandIconAriaLabelledby;
+        // If there is no explicit aria-label, calculate default accName attribute for expandIcon button,
+        // using the following steps:
+        // 1. If there is an aria-label, it is "Open [aria-label]"
+        // 2. If there is an aria-labelledby, it is "Open [aria-labelledby target]" (using aria-labelledby + ids)
+        // 3. If there is no aria-label/ledby attr, it falls back to "Open"
+        // We can't fall back to a label/htmlFor name because of https://github.com/w3c/accname/issues/179
+        const defaultOpenString = 'Open'; // this is english-only since it is the fallback
+        if (!hasExpandLabel) {
+          if (ariaLabelledBy) {
+            const chevronId = expandIconId ?? `${tagPickerId}-chevron`;
+            const chevronLabelledBy = `${chevronId} ${ariaLabelledBy}`;
+
+            expandAriaLabel = defaultOpenString;
+            expandId = chevronId;
+            expandAriaLabelledBy = chevronLabelledBy;
+          } else if (ariaLabel) {
+            expandAriaLabel = `${defaultOpenString} ${ariaLabel}`;
+          } else {
+            expandAriaLabel = defaultOpenString;
+          }
+        }
+      }
+
+      return { expandAriaLabel, expandAriaLabelledBy, expandId };
+    },
+    [expandIconAriaLabel, expandIconAriaLabelledby, expandIconId, hasExpandIcon, tagPickerId],
+  );
+
+  const setExpandLabel = React.useCallback(() => {
+    const inputAriaLabel = triggerRef.current?.getAttribute('aria-label');
+    const inputAriaLabelledBy = triggerRef.current?.getAttribute('aria-labelledby');
+
+    const { expandAriaLabel, expandAriaLabelledBy, expandId } = getExpandLabel(inputAriaLabel, inputAriaLabelledBy);
+
+    if (expandAriaLabelledBy) {
+      expandIconRef.current?.setAttribute('aria-labelledby', expandAriaLabelledBy);
+    } else if (expandAriaLabel) {
```

Comment:
this should be a series of three `if`s, rather than `else if` (for aria-labelledby, aria-label, and id). This is because even though `aria-labelledby` will override `aria-label`, since we're actually using the self-referencing `id` in `aria-labelledby={expandId expandAriaLabelledBy}`, we do still need to set `aria-label` in addition to `aria-labelledby`